### ISSUE TITLE
Active support throws an exception when a module is included within ApplicationController

### ISSUE THREAD

**Issue by i2chris** [rails/rails#36116]
### Steps to reproduce
I'm having trouble recreating this bug that appears in a large rails project.  I have just updated my app from 5.2 to 6.0.0.rc1 and I cannot run the app because of it.

It happens when a module (that lives in app/domains/roles/authorisation) is included in Application controller:
```
class ApplicationController < ActionController::Base
  protect_from_forgery with: :exception, prepend: true
  before_action :authenticate_user!
  include ::Roles::Authorisation::ControllerMethods  <--- this line throws an exception
```
config/application.rb
``` 
   config.autoload_paths << "#{Rails.root}/app/domains"
```
app/domains/roles/authorisation/controller_methods.rb
```
module Roles
  module Authorisation
    module ControllerMethods
      extend ActiveSupport::Concern
      included do
        ... 
      end
      ...
   end
 end
end
```

### Expected behavior
A module should be included in ApplicationController

### Actual behavior

It throws an exception when it tries to include it:
```
#<TypeError: no implicit conversion of true into String>
/home/chrisr/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.0.rc1/lib/active_support/dependencies.rb:468:in `join'
/home/chrisr/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.0.rc1/lib/active_support/dependencies.rb:468:in `autoload_module!'
/home/chrisr/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.0.rc1/lib/active_support/dependencies.rb:531:in `load_missing_constant'
/home/chrisr/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:43:in `load_missing_constant'
/home/chrisr/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.0.rc1/lib/active_support/dependencies.rb:206:in `const_missing'
/home/chrisr/theapp/app/controllers/application_controller.rb:4:in `<class:ApplicationController>'
/home/chrisr/theapp/app/controllers/application_controller.rb:1:in `<main>'
```

Somehow base_path is false, rather than a string, on line 468 of active_support/dependencies.rb:
```
    def autoload_module!(into, const_name, qualified_name, path_suffix)
      return nil unless base_path = autoloadable_module?(path_suffix)
      mod = Module.new
      into.const_set const_name, mod
->       log("constant #{qualified_name} autoloaded (module autovivified from #{File.join(base_path, path_suffix)})")
      autoloaded_constants << qualified_name unless autoload_once_paths.include?(base_path)
      autoloaded_constants.uniq!
      mod
```

### System configuration
**Rails version**:
gem 'rails', '~> 6.0.0.rc1'

**Ruby version**:
2.6.3

**Comment by i2chris** [comment]
Closing.  Somehow I fixed this and I'm not 100% sure how.  I'll update if I find the exact reason.

**Comment by smidwap** [comment]
I got the same error when upgrading to Rails 6.0 and fixed it by updating this line in config/environments/application.rb:

```
config.load_defaults 5.1
```

to

```
config.load_defaults 6.0
```

**Comment by one0fnine** [comment]
guys, just update bootsnap gem to fix that

**Comment by rmacklin** [comment]
To expand a bit on https://github.com/rails/rails/issues/36116#issuecomment-568717428, there was an issue in bootsnap's implementation of `autoloadable_module?` which would lead to this error - it was fixed here: https://github.com/Shopify/bootsnap/pull/199 (available in bootsnap >= 1.3.2)
